lib/readable.js

```diff
@@ -96,29 +99,47 @@ class KinesisReadable extends Readable {
       ShardIterator: shardIterator,
       Limit: 10000, // https://github.com/awslabs/amazon-kinesis-client/issues/4#issuecomment-56859367
     }
+    let keepGoing = true
     let data
     try {
       data = await this.client.getRecords(params).promise()
     } catch (err) {
       this.emit('error', err)
-      return
-    }
-    if (data.MillisBehindLatest > 60 * 1000) {
-      this.logger.warn('readShard behind by %d milliseconds', data.MillisBehindLatest)
-    }
-    data.Records.forEach((x) => this.push(this.options.parser(x.Data)))
-    if (data.Records.length) {
-      this.emit('checkpoint', data.Records[data.Records.length - 1].SequenceNumber)
+      // if the error says it can be retried then keep going
+      keepGoing = err.retryable
```

Comment:
This was so far mostly useful when developing but I can see it in production as well. If the network connection goes down then the request will fail with retryable true. Maxretries would get you some way but it would still fail eventually if the connection stays down for longer.